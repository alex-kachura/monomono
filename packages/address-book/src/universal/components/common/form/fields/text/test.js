import React from 'react';
import { render } from '@testing-library/react';
import { renderProviders } from '@oneaccount/react-foundations/lib/test-utils';
import { DefaultThemeProvider } from '@beans/theme';
import Text from './';
import { Formik } from '@oneaccount/react-foundations';

const props = {
  id: 'test',
  name: 'test',
  label: 'Test',
  placeholder: 'Test',
  required: true,
};

function renderText({ value = '', error, textProps }) {
  return render(
    renderProviders({
      appConfig: {
        getLocalePhrase: (key) => key,
      },
      children: (
        <DefaultThemeProvider>
          <Formik initialValues={{ test: value }} initialErrors={{ test: error }}>
            <Text {...textProps} />
          </Formik>
        </DefaultThemeProvider>
      ),
    }),
  );
}

describe('[Component: Text]', () => {
  describe('[State: Default]', () => {
    it('should render correctly', () => {
      const component = renderText({ value: '', textProps: props });

      expect(component.asFragment()).toMatchSnapshot();
    });
  });

  describe('[State: Initial Value]', () => {
    it('should render correctly', () => {
      const component = renderText({
        value: 'Some content',
        textProps: props,
      });

      expect(component.asFragment()).toMatchSnapshot();
    });
  });

  describe('[State: Error]', () => {
    it('should render correctly', () => {
      const component = renderText({
        value: '',
        error: 'Error on field',
        textProps: props,
      });

      expect(component.asFragment()).toMatchSnapshot();
    });
  });
});
