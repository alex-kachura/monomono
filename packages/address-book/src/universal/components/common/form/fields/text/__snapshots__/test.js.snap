// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component: Text] [State: Default] should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa cZgcHL styled__Container-sc-1bf1p22-0 hTDNaU"
  >
    <label
      class="styled__StyledLabel-sc-28yvb4-0 bPkari"
      for="test"
    >
      Test 
    </label>
    <div
      class="base-components__BaseElement-sc-150tvch-0 beans-form-group__children styled__FormGroupChildren-wt4ift-2 bNvdOT"
    >
      <input
        aria-disabled="false"
        class="styled__Container-sc-1bf1p22-0 eZwXEO"
        id="test"
        name="test"
        placeholder="Test"
        required=""
        type="text"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`[Component: Text] [State: Error] should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa cZgcHL styled__Container-sc-1bf1p22-0 hTDNaU"
  >
    <label
      class="styled__StyledLabel-sc-28yvb4-0 BlFly"
      for="test"
    >
      Test 
    </label>
    <div
      class="base-components__BaseElement-sc-150tvch-0 beans-form-group__children styled__FormGroupChildren-wt4ift-2 bNvdOT"
    >
      <input
        aria-describedby="test-error"
        aria-disabled="false"
        class="styled__Container-sc-1bf1p22-0 hIxSdw"
        id="test"
        name="test"
        placeholder="Test"
        required=""
        type="text"
        value=""
      />
    </div>
    <p
      class="base-components__BaseElement-sc-150tvch-0 beans-form-group__message beans-form-group__error-message styled__Message-wt4ift-3 jcJuKh typography___styledElement-sc-186l0ce-0 dmJyuB"
      id="test-error"
    >
      Error on field
    </p>
  </div>
</DocumentFragment>
`;

exports[`[Component: Text] [State: Initial Value] should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa cZgcHL styled__Container-sc-1bf1p22-0 hTDNaU"
  >
    <label
      class="styled__StyledLabel-sc-28yvb4-0 bPkari"
      for="test"
    >
      Test 
    </label>
    <div
      class="base-components__BaseElement-sc-150tvch-0 beans-form-group__children styled__FormGroupChildren-wt4ift-2 bNvdOT"
    >
      <input
        aria-disabled="false"
        class="styled__Container-sc-1bf1p22-0 eZwXEO"
        id="test"
        name="test"
        placeholder="Test"
        required=""
        type="text"
        value="Some content"
      />
    </div>
  </div>
</DocumentFragment>
`;
